### PR TITLE
Switch WT and OpenConsole to the Segment Heap

### PR DESCRIPTION
**Pull request by DHowett**
To quote an internal wiki:

> It generally provides improved footprint and performance over the
> existing default heap for native win32 applications.

It is apparently the default heap on ARM64, and for all UWPs.

This heap has different allocation and compaction characteristics.
I am not sure how it will impact terminal.

### CODE REVIEW

File: src/host/exe/openconsole.exe.manifest [hunk 1]
```diff
@@ -15,6 +15,7 @@
     <application xmlns="urn:schemas-microsoft-com:asm.v3">
         <windowsSettings>
             <longPathAware xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">true</longPathAware>
+            <heapType xmlns="http://schemas.microsoft.com/SMI/2020/WindowsSettings">SegmentHeap</heapType>
             <!--Enable longPath support-->
```
Comment (DHowett) [inline src/host/exe/openconsole.exe.manifest:19]: Hey, this comment shouldn't even exist.

```suggestion
```

**Review by lhecker — APPROVED** [review]
No significant difference in performance. Approx. -1MB Private WS for OpenConsole.exe.

**Review by carlos-zamora — APPROVED** [review]
lgtm!

**Comment by zadjii-msft** [comment]
(this is the PR we were thinking about that may have stealth-introduced the non-zero-initialization)

**Pull request merged**